src/Common/Helpers/ServiceBusHelper.cs

```diff
@@ -4757,9 +4759,16 @@ public string GetMessageText(BrokeredMessage messageToRead, bool useAscii, out B
                 return null;
             }
             var inboundMessage = messageToRead.Clone();
+            bool gzipDecompress = false;
             try
             {
                 stream = inboundMessage.GetBody<Stream>();
+                if (messageToRead.Properties.ContainsKey("Content-Encoding"))
+                {
+                    var encoding = inboundMessage.Properties["Content-Encoding"].ToString();
+                    if (encoding == "gzip")
+                        gzipDecompress = true;
```

Comment:
minor suggestion, not critical
```suggestion
                    gzipDecompress = encoding == "gzip";
```